src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -514,16 +514,10 @@ private InstructorFeedbackResultsGroupByParticipantPanel buildGroupByParticipant
                 .setProfilePictureLink(getProfilePictureIfEmailValid(secondaryParticipantIdentifier));
             
             if (!viewType.isPrimaryGroupingOfGiverType()) {
-                boolean isStudent = bundle.roster.getStudentForEmail(secondaryParticipantIdentifier) != null;
-                boolean isVisibleTeam = isTeamVisible(secondaryParticipantDisplayableName);
                 String sectionName = bundle.getSectionFromRoster(secondaryParticipantIdentifier);
-                boolean isAllowedToModerate = (isStudent || isVisibleTeam) 
-                                           && instructor.isAllowedForPrivilege(
-                                                  sectionName, feedbackSessionName, 
-                                                  Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS);
-                boolean isShowingModerationButton = (isStudent || isVisibleTeam) 
-                                                 && isAllowedToModerate;
-                secondaryParticipantPanel.setModerationButton(isShowingModerationButton
+                boolean isAllowedToModerate = isAllowedToModerate(instructor, sectionName, feedbackSessionName);
```

Comment:
Perhaps the original variable names there were not good. But if a team is anonymous, then it would be false. <code>isStudent || isVisibleTeam || isInstructor</code> looks more correct